### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by jgreenQA**

### CODE REVIEW

**Comment by tvaidotas** [comment]
looks good

File: src/main/scala/rockpaperscissors/Move.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    override def toString: String = name
+  }
+
+  case object ROCK      extends Move("rock", List("scissors", "lizard"), List("paper", "spock"))
+  case object PAPER     extends Move("paper", List("rock", "spock"), List("scissors", "lizard"))
+  case object SCISSORS  extends Move("scissors", List("paper", "lizard"), List("rock", "spock"))
+  case object LIZARD    extends Move("lizard", List("paper", "spock"), List("rock", "scissors"))
+  case object SPOCK     extends Move("spock", List("scissors", "rock"), List("lizard", "paper"))
+
+  /*case object ROCK      extends Move("rock", List(Move.SCISSORS, Move.LIZARD), List(Move.PAPER, Move.SPOCK))
```
Comment (tvaidotas) [inline src/main/scala/rockpaperscissors/Move.scala]: is this needed?

File: src/main/scala/rockpaperscissors/RockPaperScissors.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    var players = ArrayBuffer[Player]()
+
+    if (playerName.isEmpty) {
+      for (i <- 1 to playersAmount) players += new Player(s"AI$i")
+    } else {
+      players += new Player(playerName)
+      for (i <- 1 until playersAmount) players += new Player(s"AI$i")
+    }
+
+    while (players.length > 1) { //for (i <- 1 to rounds if players.length > 1) {
```
Comment (tvaidotas) [inline src/main/scala/rockpaperscissors/RockPaperScissors.scala]: is this comment needed?

File: src/main/scala/rockpaperscissors/RockPaperScissors.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      println(s"\nRound $currRound:")
+
+      chooseMoves(players)
+
+      players.foreach(p => println(s"${p.getName} chose: ${p.getCurrMove}"))
+
+      players = determineWinner(players)
+
+      //println(s"\nAmount of players: ${players.length}")
```
Comment (tvaidotas) [inline src/main/scala/rockpaperscissors/RockPaperScissors.scala]: is this comment needed?

File: src/main/scala/rockpaperscissors/RockPaperScissors.scala [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  }
+
+  def chooseMoves(players: ArrayBuffer[Player]): Unit = {
+    val moves = Array('r', 'p', 's', 'l', 'k')
+
+    for (player <- players) {
+      var moveChar = ' '
+      if (player.getName.startsWith("AI") && players.length > 2) moveChar = Random.shuffle(moves.toList).head
+      else if (player.getName.startsWith("AI") && players.length == 2) println(s"Test: ${players.head.lastMove}") //moveChar = Random.shuffle(players.head.getLastMove.getLoses).head.charAt(0)
```
Comment (tvaidotas) [inline src/main/scala/rockpaperscissors/RockPaperScissors.scala]: is this comment needed?

File: src/main/scala/rockpaperscissors/RockPaperScissors.scala [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    var winningPlayers = ArrayBuffer[Player]()
+    var playedMoves = ArrayBuffer[Move]()
+
+    for (currentPlayer <- players) playedMoves += currentPlayer.getCurrMove
+
+    if (playedMoves.distinct.length != 2) {
+      winningPlayers = players
+    } else {
+      for (comparisonPlayer <- players; currentPlayer <- players) {
+        //println(s"\nComp = ${comparisonPlayer.getName} : ${comparisonPlayer.getCurrMove}")
```
Comment (tvaidotas) [inline src/main/scala/rockpaperscissors/RockPaperScissors.scala]: are these comments needed?

File: src/main/scala/rockpaperscissors/Move.scala (hunk 1 above)
Comment (jgreenQA) [reply src/main/scala/rockpaperscissors/Move.scala]: Not at all, it was just there while I fixed it. I'll remove them now

**Comment by tvaidotas** [comment]
Looks good

**Pull request merged**